test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala

```diff
@@ -30,6 +30,57 @@ class BaseCharsetTest(val charset: Charset) {
   protected def testDecode(in: ByteBuffer)(
       outParts: OutPart[CharBuffer]*): Unit = {
 
+    def testDecodeVsSteppedDecode(malformedAction: CodingErrorAction,
+        unmappableAction: CodingErrorAction) = {
+
+      val in1 = in.duplicate
+      val in2 = in.duplicate
+
+      in2.limit(in2.position)
+
+      val d1 = charset.newDecoder()
+      d1.onMalformedInput(malformedAction)
+      d1.onUnmappableCharacter(unmappableAction)
+
+
+      val r1 = Try(d1.decode(in1))
+      val out2 = CharBuffer.allocate(in.capacity * 2)
+      val d2 = charset.newDecoder()
+      d2.onMalformedInput(malformedAction)
+      d2.onUnmappableCharacter(unmappableAction)
+
+      def increaseBuffer(bb: ByteBuffer) = (bb.limit < bb.capacity) && (bb.limit(bb.limit + 1) != null)
+
+      val r2 = Try {
+        var result: CoderResult = CoderResult.UNDERFLOW
+        while(increaseBuffer(in2) && result.isUnderflow){
+          result = d2.decode(in2, out2, false)
+        }
+        if(result.isError) result.throwException
+
+        result = d2.decode(in2, out2, true)
+        if(result.isError) result.throwException
+        result = d2.flush(out2)
+        if(result.isError) result.throwException
+        out2.flip
+        out2
+      }
+
+      (r1, r2) match {
+        case (Success(b1), Success(b2)) => assert(b1.toString == b2.toString)
+        case (Failure(err1), Failure(err2)) => {
+          (err1, err2) match {
+            case (m1: MalformedInputException, m2: MalformedInputException) =>
+              assert(m1.getInputLength == m2.getInputLength)
+            case (u1: UnmappableCharacterException, u2: UnmappableCharacterException) =>
+              assert(u1.getInputLength == u2.getInputLength)
+            case _ => assert(false)
```

Comment:
Consider using `assertSame` for better error messages, like it's done [in `testOneConfig`](https://github.com/scala-js/scala-js/blob/ed20d0dac5d92f4a84edf5837054bae70aecb021/test-suite/shared/src/test/scala/org/scalajs/testsuite/niocharset/BaseCharsetTest.scala#L94-L96).